lib/AppInfo/Application.php

```diff
@@ -83,6 +90,38 @@ public function register() {
 				Util::addScript('terms_of_service', 'terms_of_service_public');
 			}
 		}
+
+		// register Sabre plugin
+		/** @var IEventDispatcher $eventDispatcher */
+		$eventDispatcher = $server->query(IEventDispatcher::class);
+		$eventDispatcher->addListener('OCA\DAV\Connector\Sabre::addPlugin', function (SabrePluginEvent $event) use ($server) {
+			$eventServer = $event->getServer();
+
+			if ($eventServer !== null) {
+				// We have to register the CheckPlugin here and not info.xml,
+				// because info.xml plugins are loaded, after the
+				// beforeMethod:* hook has already been emitted.
+				$plugin = $this->getContainer()->query(CheckPlugin::class);
+				if ($server->getUserSession()->getUser() !== null) {
+					$checker = $server->query(Checker::class);
+				} else {
+					// Here, if we get a Checker instance with `$server->query(Checker::class)`
+					// its userId attribute is null and ToS are considered not signed
+					$checker = new Checker(
+						// if there is no user session, how do we get the user ID used to authenticate when doing a DAV request?
+						$_SERVER['PHP_AUTH_USER'],
```

Comment:
Maybe @rullzer knows more about that but probably the sabre auth backend was not triggered at this point. Maybe you can just initialize the checker within the plugin event handlers at https://github.com/nextcloud/terms_of_service/pull/487/files#diff-ed623a65728c51fffc2e375e06520caa5ea2d45373f750baae600cfad16ce100R55-R56

Reply:
Thanks! I'll try that.

Reply:
UserId is indeed there if the Checker is created in the event handler. Thanks.